applications/IgaApplication/custom_modelers/nurbs_geometry_modeler.cpp

```diff
@@ -175,22 +175,25 @@ namespace Kratos
 
         // Set up weights. Required in case no refinement is performed.
         Vector WeightsRefined(points.size(),1.0);
-
-        // Add geometry to model part
-        if( mParameters.Has("geometry_name") ){
-            p_surface_geometry->SetId(mParameters["geometry_name"].GetString());
-        } else {
-            const SizeType number_of_geometries = r_model_part.NumberOfGeometries();
-            SizeType last_geometry_id = 0;
-            if( number_of_geometries > 0 ){
-                for( auto it = r_model_part.GeometriesBegin(); it!= r_model_part.GeometriesEnd(); ++it){
-                    last_geometry_id = it->Id();
+        
+        // In some cases there is no need of add the surface geometry, add_surface_to_model_part is true by default
+        if (add_surface_to_model_part) {
+            // Add geometry to model part
+            if( mParameters.Has("geometry_name") ){
+                p_surface_geometry->SetId(mParameters["geometry_name"].GetString());
+            } else {
+                const SizeType number_of_geometries = r_model_part.NumberOfGeometries();
+                SizeType last_geometry_id = 0;
+                if( number_of_geometries > 0 ){
+                    for( auto it = r_model_part.GeometriesBegin(); it!= r_model_part.GeometriesEnd(); ++it){
+                        last_geometry_id = it->Id();
+                    }
                 }
```

Comment:
```suggestion
                const SizeType last_geometry_id = r_model_part.NumberOfGeometries()
                                                  ? r_model_part.Geometries().Geometries().back().Id()
                                                  : 0;
```

I know, it looks stupid. In fact, it **is** stupid, but only because the geometry container is still weird.